carma_ros2_utils/include/carma_ros2_utils/timers/testing/TestTimerFactory.hpp

```diff
@@ -37,19 +38,21 @@ class TestTimerFactory : public TimerFactory
   ~TestTimerFactory();
 
   /**
-   * @brief Set the rclcpp clock interface which will be used by this test timer
+   * @brief Set all previously created timer's time to the given time
    * 
-   * @param clock_interface The interface to set
+   * @param current_time 
    */ 
-  void setClockInterface(rclcpp::node_interfaces::NodeClockInterface::SharedPtr clock_interface);
+  void setNow(const rclcpp::Time& current_time);
 
   //// Overrides
   std::unique_ptr<Timer> buildTimer(uint32_t id, rclcpp::Duration duration,
                                     std::function<void()> callback, bool oneshot = false,
                                     bool autostart = true) override;
-
+  rclcpp::Time now();
```

Comment:
I think GeofenceScheduler needs current time from the Clock, and only place it has the Clock saved is in timerFactory. So I was thinking we either need `rclcpp::Clock::SharedPtr getClock()` or return `rclcpp::Time now()` function, and that function need to be in the template TimerFactory class to work for both real life and unit tests. So I thought instead of having `TestClock` extend `Clock` in unit test, I am currently returning only now(). Let me know if it can be done differently.

Reply:
Ok, that's a fair argument. I'm good with this then.